### PR TITLE
[SPARK-20932][ML]CountVectorizer support handle persistence

### PR DESCRIPTION
**Pull request by zhengruifeng**
## What changes were proposed in this pull request?
unpersist RDDs `input` & `wordCounts` after computation

## How was this patch tested?
existing tests

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/feature/CountVectorizer.scala [hunk 1]
```diff
@@ -154,13 +155,19 @@ class CountVectorizer @Since("1.5.0") (@Since("1.5.0") override val uid: String)
   override def fit(dataset: Dataset[_]): CountVectorizerModel = {
     transformSchema(dataset.schema, logging = true)
     val vocSize = $(vocabSize)
+
+    val handlePersistence = $(minDF) < 1.0 &&
+      dataset.storageLevel == StorageLevel.NONE
```
Comment (hhbyyh) [inline mllib/src/main/scala/org/apache/spark/ml/feature/CountVectorizer.scala]: I understand your logic, but I don't think it's necessary to make it complicated. How about we just add the unpersist for both input and wordCounts.

**Comment by HyukjinKwon** [comment]
@zhengruifeng Would you answer or address the review comments above?

**Comment by zhengruifeng** [comment]
@hhbyyh @HyukjinKwon  Sorry to reply late. 
I think it may be better to use a special logic if it is more efficient in performance. 
What is your opinion? @yanboliang @HyukjinKwon

**Comment by HyukjinKwon** [comment]
I don't know ML as much as reviewing this. I just wanted to be sure if it is in progress in any way.

**Pull request closed without merging**